### PR TITLE
[MDS-5606] Typescript conversions within Summary and Variance sections

### PR DESCRIPTION
**Pull request by asinn134**
## Objective 

[MDS-5606](https://bcmines.atlassian.net/browse/MDS-5606)

- Some conversions to Typescript within the Summary and Variance section of the Core codebase

### CODE REVIEW

**Review by henryoforeh-dev — DISMISSED** [review]
Nice work!

File: services/common/src/interfaces/common/match.interface.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+export interface IMatch {
```
Comment (taraepp) [inline services/common/src/interfaces/common/match.interface.ts]: I would say this is actually not necessary. In MineSummary, for example, you can replace
`const { id } = this.props.match.params`
with
`const { id } = useParams();`
and just get rid of the prop

File: services/core-web/src/components/mine/Summary/MineSummary.tsx [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+  </Col>
+);
+
+// Since end date is stored at yyyy-mm-dd, comparing current Date() to
+// the the start of the next day ensures appointments ending today are displayed.
+const isActive = (pr) =>
+  (!pr.end_date ||
+    moment(pr.end_date)
+      .add(1, "days")
+      .toDate() > new Date()) &&
```
Comment (taraepp) [inline services/core-web/src/components/mine/Summary/MineSummary.tsx]: we don't need to convert moment to a date.

moment(date).add(1 days).isBefore() <-- isBefore used without an argument will compare to the equivalent of new Date(). There's a few variations on that function like isAfter if I'm getting my < and > mixed up. Would do something similar below with the Date.parse.

File: services/core-web/src/components/mine/Summary/MineWorkInformation.tsx [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        </Col>
+      </Row>
+    </List.Item>
+  );
+
+  const dataSource = showAll
+    ? props.mineWorkInformations
+    : !isEmpty(props.mineWorkInformations)
+    ? [props.mineWorkInformations[0]]
+    : [];
```
Comment (taraepp) [inline services/core-web/src/components/mine/Summary/MineWorkInformation.tsx:196]: now I might be getting picky but did the linter run? I was wondering how this syntax could be real until I noticed that the indentation is wacky. 😂

**Review by taraepp — COMMENTED** [review]
There's a few little things carried over that I think would be worth fixing but overall looks great!

File: services/core-web/src/components/mine/Summary/MineWorkInformation.tsx (hunk 3 above)
Comment (asinn134) [reply services/core-web/src/components/mine/Summary/MineWorkInformation.tsx:196]: The linter definitely ran here. On my laptop the indentation looks like this

```
  const dataSource = showAll
    ? props.mineWorkInformations
    : !isEmpty(props.mineWorkInformations)
      ? [props.mineWorkInformations[0]]
      : [];
```
but the linter changes it to what we see in the pr.

File: services/core-web/src/components/mine/Summary/MineSummary.tsx (hunk 2 above)
Comment (asinn134) [reply services/core-web/src/components/mine/Summary/MineSummary.tsx]: Nice! I made the change it looks a lot cleaner now.

File: services/common/src/interfaces/common/match.interface.ts (hunk 1 above)
Comment (asinn134) [reply services/common/src/interfaces/common/match.interface.ts]: Yah looking at this again the IMatch really isn't necessary. It has been removed now and useParams() is used within MineSummary instead.

File: services/core-web/src/components/mine/Summary/MineWorkInformation.tsx (hunk 3 above)
Comment (taraepp) [reply services/core-web/src/components/mine/Summary/MineWorkInformation.tsx:196]: _Gross_. But I accept this explanation. 👍

File: services/common/src/interfaces/common/match.interface.ts (hunk 1 above)
Comment (taraepp) [reply services/common/src/interfaces/common/match.interface.ts]: Nice!
I see the tests are giving you trouble- we've got a few examples of mocking this in MineIncident.spec, noticeOfDeparture.spec, and ProjectSummaryPage.spec

Comment (asinn134) [reply services/common/src/interfaces/common/match.interface.ts]: Thanks for mentioning the examples, that helped a lot

**Review by henryoforeh-dev — DISMISSED** [review]

File: services/common/src/interfaces/permits/permit.interface.ts [hunk 4]
```diff
@@ -23,4 +23,5 @@ export interface IPermit {
   site_properties: IMineType;
   permit_prefix: string;
   mine_guid?: string;
+  amendments?: Partial<IPermitAmendment>[];
```
Comment (taraepp) [inline services/common/src/interfaces/permits/permit.interface.ts]: Wait, we have `amendments` and `permit_amendments`.

Comment (asinn134) [reply services/common/src/interfaces/permits/permit.interface.ts]: Oh whoops I forgot to remove this after some testing. Thanks for catching this.

**Review by taraepp — APPROVED** [review]

**Review by henryoforeh-dev — APPROVED** [review]

**Pull request merged**